src/main/java/org/zalando/nakadi/service/subscription/StreamingContext.java

```diff
@@ -264,11 +264,19 @@ public ObjectMapper getObjectMapper() {
     private void rebalance() {
         if (null != sessionListSubscription) {
             // This call is needed to renew subscription for session list changes.
-            sessionListSubscription.getData();
+            final List<String> newSessions = sessionListSubscription.getData();
+            final String newHash = ZkSubscriptionClient.Topology.calculateSessionsHash(newSessions);
```

Comment:
Why isn't it enough to compare topologies by partitions, session and event types? It seems like additional logic is on purpose

Reply:
The problem with it is that it may happen that it is completely normal to have sessions that do not own any partitions. Therefore the list of sessions is managed separately. But I will fix the bug with recalculation of sessions hash on delayed call that we discussed offline